packages/core/src/storage/IndyStorageService.ts

```diff
@@ -172,13 +172,17 @@ export class IndyStorageService<T extends BaseRecord> implements StorageService<
 
   /** @inheritDoc */
   public async delete(record: T) {
+    await this.deleteById(record.type, record.id)
+  }
+
+  /** @inheritDoc */
+  public async deleteById(recordType: string, id: string): Promise<void> {
```

Comment:
I would rather keep it as a record class here, like we have with other types. 

So:

```ts
public async deleteById(recordClass: BaseRecordConstructor<T>, id: string): Promise<T> {
```

Reply:
Yes, I thought on avoiding some code duplication but didn't notice that it would imply to change the API. I'm wondering if there's a way of getting the best of both worlds but anyway it'll be fine if we keep as it was before. Sorry @tipusingh !

Reply:
Ah I see now why we had to change both. Yeah also hadn't noticed that.

We could achieve what we want using something like `record: { type: string }` which means we'll always have the type. Or we should make a private method that takes the string type that both public method will use.